### PR TITLE
EVG-18987 Install react-virtuoso and create ADR for virtualization migration

### PR DESCRIPTION
**Pull request by khelif96**
EVG-18987

### Description 
Installs the react-virtuoso library and creates a WIP ADR that will be updated as this refactor progresses

### Testing 
Tested to see if I could import react-virtuoso without running into any vite problems.

### CODE REVIEW

File: docs/decisions/0003-migrating-to-react-virtuoso.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Migrating Parsley Virtualization to react-virtuoso
+
+* status: proposed
+* date: 2022-02-22
```
Comment (bsamek) [inline docs/decisions/0003-migrating-to-react-virtuoso.md:4]:
```suggestion
* date: 2023-02-22
```

File: docs/decisions/0003-migrating-to-react-virtuoso.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+* authors: Mohamed Khelif
+
+## Context and Problem Statement
+
+Parsley currently uses the [react-virtualized](https://github.com/bvaughn/react-virtualized) library to render log files. This library was the only library we could find that can support a continuous virtualized view capable of rendering over 800k lines, which is often the case with some log files. Unfortunately it has many problems, some of which include:
+
+* It is no longer maintained by its owner.
+* It is unable to maintain a scroll position when scrolling to previously unvisited lines due to some buggy scroll position compensation logic. (Which ironically is probably the thing that lets it render so many lines). This causes the log file to "jump" dramatically.
+  * It needs to maintain a cache of the sizes of each log line.
```
Comment (bsamek) [inline docs/decisions/0003-migrating-to-react-virtuoso.md:13]: indentation?

FYI there's a build-in markdown preview mode in vs code: `cmd-k, v`

File: docs/decisions/0003-migrating-to-react-virtuoso.md (hunk 1 above)
Comment (khelif96) [reply docs/decisions/0003-migrating-to-react-virtuoso.md:4]: oops i'm still stuck in 2022 gonna take me a few months to adjust :/

File: docs/decisions/0003-migrating-to-react-virtuoso.md (hunk 2 above)
Comment (khelif96) [reply docs/decisions/0003-migrating-to-react-virtuoso.md:13]: This is intentional it is meant to be a sub-point. It reflects the contents of the ticket. https://jira.mongodb.org/browse/EVG-18993

Comment (bsamek) [reply docs/decisions/0003-migrating-to-react-virtuoso.md:13]: Ah gotcha, sorry!

**Review by bsamek — APPROVED** [review]

**Review by minnakt — APPROVED** [review]
yay! feel free to edit my edits! 😳

**Pull request merged**